src/Android/Xamarin.Android/Samples/Data/StatsQueryGroupAndSort/StatsQueryGroupAndSort.cs

```diff
@@ -0,0 +1,1000 @@
+
+            // Begin a transaction to show a UI fragment (stats definitions dialog)
+            FragmentTransaction trans = FragmentManager.BeginTransaction();
+            statsDefDialog.Show(trans, "stats_defs");
+        }
+
+        private void ShowGroupFields(object sender, EventArgs e)
+        {
+            // Create a dialog for choosing fields to group results with
+            var groupFieldsDialog = new GroupFieldsDialog(_groupByFields);
+
+            // Handle the dialog closing event to read the selected fields
+            groupFieldsDialog.GroupFieldDialogClosed += (s, args) =>
+            {
+                // Update the dictionary of group fields from the dialog
+                _groupByFields = args;
+
+                // Get the current list of group fields and create/update the sort field choices
+                // (only fields selected for grouping can be used to order results)
+                var currentGroupFields = _groupByFields.Where((f) => f.Value == true);
+
+                // Loop through the group fields
+                foreach (var f in currentGroupFields)
+                {
+                    // Check if this field is missing from the current sort field options
+                    var existingOption = _orderByFields.Find((opt) => opt.OrderInfo.FieldName == f.Key);
+                    if (existingOption == null)
+                    {
+                        // If the field is missing, create a new OrderFieldOption and add it to the list
+                        existingOption = new OrderFieldOption(false, new OrderBy(f.Key, SortOrder.Ascending));
+                        _orderByFields.Add(existingOption);
+                    }
+                }
+
+                // Also make sure to remove any 'order by' fields that were removed from the 'group by' list
+                for (var i = _orderByFields.Count - 1; i >= 0; i--)
+                {
+                    // If the order by field is not also one of the group fields, remove it from the list
+                    var opt = _orderByFields.ElementAt(i);
+                    var existingGroupField = currentGroupFields.FirstOrDefault((f) => f.Key == opt.OrderInfo.FieldName);
+                    if (existingGroupField.Key == null)
+                    {
+                        _orderByFields.RemoveAt(i);
+                    }
+                }
+            };
+
+            // Begin a transaction to show a UI fragment (group fields dialog)
+            FragmentTransaction trans = FragmentManager.BeginTransaction();
+            groupFieldsDialog.Show(trans, "group_flds");
+        }
+
+        private void ShowOrderByFields(object sender, EventArgs e)
+        {
+            // If there are no available order fields, don't show the (empty) list
+            if (_orderByFields.Count == 0)
+            {
+                // Warn the user to choose group fields first
+                var alertBuilder = new AlertDialog.Builder(this);
```

Comment:
This is just a suggestion, but I like the pattern of having a separate 'ShowMessage' function. This way the code can be kept more consistent between samples and any repetition is avoided when multiple functions need to show messages.